api4/post_test.go

```diff
@@ -1855,48 +1855,66 @@ func TestGetPostsForChannelAroundLastUnread(t *testing.T) {
 func TestGetPost(t *testing.T) {
 	th := Setup(t).InitBasic()
 	defer th.TearDown()
+	// TODO: migrate this entirely to the subtest's client
+	// once the other methods are migrated too.
 	Client := th.Client
 
-	post, resp := Client.GetPost(th.BasicPost.Id, "")
-	CheckNoError(t, resp)
+	var privatePost *model.Post
+	th.TestForClientAndLocal(t, func(t *testing.T, c *model.Client4) {
```

Comment:
I think we can use `TestForAllClients` here. The system admin and local mode clients should behave the same. We can then drop the check that's on line 1914

Reply:
Done. Left the utility function though. In case someone needs it.